### PR TITLE
keybase: 6.0.2 -> 6.2.4; keybase-gui: 6.0.2 -> 6.2.4; backport to 23.05

### PR DESCRIPTION
**Pull request by eyeinsky**
## Description of changes

This is a backport to ~~23.11~~ (typo, correct: 23.05), both commits are already in unstable via these PRs:
- https://github.com/NixOS/nixpkgs/pull/278033
- https://github.com/NixOS/nixpkgs/pull/278346

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by teutat3s** [comment]
Looks like an update to the goModules hash is needed for this to build:
```
error: hash mismatch in fixed-output derivation '/nix/store/dqmlrxmh9rcp8ym4lj1m54d9ah2lv0fb-keybase-6.2.4-go-modules.drv':
         specified: sha256-D8b/pvmBGCnaRuf92FYgRcSSbN59Yu0CHKxAybdYjS4=
            got:    sha256-tXEEVEfjoKub2A4m7F3hDc5ABJ+R+axwX1+1j7e3BAM=
```

File: pkgs/tools/security/keybase/default.nix [hunk 1]
```diff
@@ -16,7 +16,7 @@ buildGoModule rec {
     owner = "keybase";
     repo = "client";
     rev = "v${version}";
-    sha256 = "sha256-JiYufEsoj/98An2qKdm/Uu4YHJr6ttc/VHn4kMgkuwI=";
+    hash = "sha256-z7vpCUK+NU7xU9sNBlQnSy9sjXD7/m8jSRKfJAgyyN8=";
   };
   vendorSha256 = "sha256-D8b/pvmBGCnaRuf92FYgRcSSbN59Yu0CHKxAybdYjS4=";
```
Comment (teutat3s) [inline pkgs/tools/security/keybase/default.nix:21]:
```suggestion
  vendorSha256 = "sha256-tXEEVEfjoKub2A4m7F3hDc5ABJ+R+axwX1+1j7e3BAM=";
```

**Review by teutat3s — CHANGES_REQUESTED** [review]

**Comment by eyeinsky** [comment]
@teutat3s  What is the command that you use to get this error? When in the nixpkgs repo, the following command succeeds for me `nix-build . -A keybase` (and also for `keybase-gui`). Is there anything else I could try to reproduce, maybe clear the download cache somehow?

**Comment by teutat3s** [comment]
@eyeinsky You can find this error when looking at the ofborg x86_64-linux check in this PR, then follow the [link to the full logs](https://logs.ofborg.org/?key=nixos/nixpkgs.279435&attempt_id=9ec45006-6b2e-46c9-81d7-30ba18526d41). I have not tried to reproduce this locally.

**Comment by eyeinsky** [comment]
Right, the change fixed the issue.

There is another build error for darwin, though:
```
       > In file included from /nix/store/xxbz3p03wa8wlbhb8c91ndh3f2rad1jp-apple-framework-CoreMedia/Library/Frameworks/CoreMedia.framework/Headers/CMSampleBuffer.h:56:
       > In file included from /nix/store/xxbz3p03wa8wlbhb8c91ndh3f2rad1jp-apple-framework-CoreMedia/Library/Frameworks/CoreMedia.framework/Headers/CMFormatDescription.h:27:
       > /nix/store/bvqircp5apqaqyk57r8m4z38hfbdabfq-apple-framework-AudioToolbox/Library/Frameworks/AudioToolbox.framework/Headers/AudioFormat.h:161:8: error: redefinition of 'AudioFormatListItem'
       > struct AudioFormatListItem
       >        ^
       > /nix/store/a67qmarhk5cca9a4s19r0ys3q99kyndp-apple-framework-CoreAudioTypes-11.0.0/Library/Frameworks/CoreAudioTypes.framework/Headers/CoreAudioBaseTypes.h:1412:8: note: previous definition is here
       > struct AudioFormatListItem
       >        ^
       > 1 error generated.
       For full logs, run 'nix log /nix/store/2avqqjb3svdzmnqn608vm79z2w9vx8zj-keybase-6.2.4.drv'.
```

- `CoreMedia` [appears in keybase codebase](https://github.com/search?q=org%3Akeybase%20CoreMedia&type=code)
- googling for "error: redefinition of 'AudioFormatListItem'" doesn't give anything though

**Comment by eyeinsky** [comment]
This is ready for merge from my part.

Regarding the darwin issue, is there a way to disable a package for a certain system? I don't currently have darwin machine to test and fix it.

**Comment by teutat3s** [comment]
I can reproduce the build error on `x86_64-darwin`. I tried getting some help from #macos:nixos.org on Matrix, but I guess it would also be fine to just mark `x86_64-darwin` as broken.

**Comment by teutat3s** [comment]
```
meta = {
...
  broken = stdenv.isDarwin && stdenv.isx86_64;
...
};
```

**Comment by eyeinsky** [comment]
Yup. I think many will install keybase using the official dmg anyway. Pushed the change.

**Comment by teutat3s** [comment]
To nits that might be required to get this merged:
- Rebase and squash commit 330c573212fd681723d747b658b73762647f42cc into bb71ac594d49ebe960653ef7419a4a5d58a93330
- Rephrase commit message for 0daaead8399cd2f85b3250f457d2d6d1e342fbb8 to `keybase: mark as broken on x86_64-darwin`

**Comment by eyeinsky** [comment]
Done!

**Review by teutat3s — APPROVED** [review]
Diff looks good to me, waiting for ofborg. 

Thanks for handling all the backports! 🙌🏼

**Comment by NickCao** [comment]
> This is a backport to 23.11

However this is targeting 23.05?

**Comment by teutat3s** [comment]
@NickCao that's a copy pasta leftover. The PR for 23.11 already got merged. #279429

**Comment by NickCao** [comment]
23.05 has long been EOL, but anyway.

**Comment by eyeinsky** [comment]
@NickCao Thanks! Since I was doing 23.11 backport anyway, did this one as well. Maybe helps someone who hasn't upgraded yet.

**Pull request merged**